### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by Limess**
This supersedes https://github.com/Financial-Times/prometheus-biz-ops-service-discovery/pull/7.

Added unit tests for the 3 files with core logic.

Didn't add any integration tests, which may have been useful but would probably have been straight duplicates.

### CODE REVIEW

File: internal/servicediscovery/servicediscovery_test.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+			bizOpsError:   nil,
+			expectedErr:   errors.New("processed healthchecks were empty"),
+		},
+		"empty healthchecks biz-ops response should return an error": {
+			bizOpsResponse: newGraphQLResponse([]Healthcheck{}),
+			expectedWrite:  "",
+			bizOpsError:    nil,
+			expectedErr:    errors.New("returned healthchecks were empty"),
+		},
+		"with graphql error should retur an error": {
```
Comment (lucas42) [inline internal/servicediscovery/servicediscovery_test.go]: nit return

File: internal/api/bizops_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+func TestBizOpsApiHandler(t *testing.T) {
+	testCases := map[string]struct {
+		query                  string
+		bizOpsResponse         string
+		bizOpsResponseCode     int
+		expectedApiKey         string
+		expectedParsedResponse interface{}
+		expectedErr            error
+	}{
+		"invalid query should return an error": {
```
[the remark this answers is not in the document]
Comment (lucas42) [reply internal/api/bizops_test.go]: Can we rename it to something a bit clearer about the actual case it's testing?

**Comment by Limess** [comment]
@lucas42 fixed your comments that I didn't reply to.
